### PR TITLE
Load data/overscan sections from headers

### PR DESCRIPTION
**Pull request by rcooke-ast**
This PR adds the functionality to load keyword information from header files (addresses issue #260).

This PR includes all of the changes currently in PR #282, and should **_only_** be considered after PR #282 has been merged.

This PR passes the development suite.

### CODE REVIEW

**Comment by profxj** [comment]
@rcooke-ast -- Can you pull master into this branch and 
then I'll review it?  Thanks.

**Comment by rcooke-ast** [comment]
Sounds good - I'll just make one quick change... more soon.

**Comment by rcooke-ast** [comment]
By the way, this now passes the dev suite for me. Perhaps it would be good to confirm that your machine passes the devsuite too.

**Comment by profxj** [comment]
Running now.  On the master branches of arclines and Development.

File: pypit/ararc.py [hunk 1]
```diff
@@ -130,10 +130,13 @@ def fit_arcspec(xarray, yarray, pixt, fitp):
...
-        popt = arutils.func_fit(xarray[pmin:pmax], yarray[pmin:pmax], "gaussian", 3)
-        ampl[p] = popt[0]
-        cent[p] = popt[1]
-        widt[p] = popt[2]
+        try:
+            popt = arutils.func_fit(xarray[pmin:pmax], yarray[pmin:pmax], "gaussian", 3)
+            ampl[p] = popt[0]
+            cent[p] = popt[1]
+            widt[p] = popt[2]
+        except RuntimeError:
```
Comment (profxj) [inline pypit/ararc.py]: should we at least throw a warning here?

File: pypit/arparse.py [hunk 2]
```diff
@@ -3758,7 +3790,7 @@ def get_nmbr_name(anmbr=None, bnmbr=None, cnmbr=None):
     return "_".join(cspl)
 
 
-def load_sections(string):
+def load_sections(string, fmt_iraf=True):
```
Comment (profxj) [inline pypit/arparse.py]: Add doc describing fmt_iraf

File: pypit/arparse.py [hunk 3]
```diff
@@ -4004,7 +4046,7 @@ def key_int(v):
     return v
 
 
-def key_keyword(v):
+def key_keyword(v, force_format=True):
```
Comment (profxj) [inline pypit/arparse.py]: add doc on force_format

File: pypit/data/settings/archive/settings.2017-11-17.apf_levy [hunk 4]
```diff
@@ -18,8 +18,8 @@ det01 saturation 65535.0              # The detector Saturation level
 det01 nonlinear 0.8                  # Percentage of detector range which is linear (i.e. everything above nonlinear*saturation will be flagged as saturated)
 det01 numamplifiers 1                 # Number of amplifiers
 det01 dataext01 0                     # Extension number of the data
-det01 datasec01 [0:4608,0:2047]       # Either the data sections or the header keyword where the valid data sections can be obtained
-det01 oscansec01 [0:4608,2049:2080]   # Either the overscan sections or the header keyword where the valid overscan sections can be obtained
+det01 datasec01 [1:2047,1:4608]       # Either the data sections or the header keyword where the valid data sections can be obtained
```
Comment (profxj) [inline pypit/data/settings/archive/settings.2017-11-17.apf_levy]: Maybe comment that the formatting  is 'IRAF'.

Here and throughout.

**Comment by profxj** [comment]
Had to add a Python 3 fix to pass Development.
All good now

File: pypit/ararc.py (hunk 1 above)
Comment (rcooke-ast) [reply pypit/ararc.py]: done

File: pypit/arparse.py (hunk 2 above)
Comment (rcooke-ast) [reply pypit/arparse.py]: done

File: pypit/arparse.py (hunk 3 above)
Comment (rcooke-ast) [reply pypit/arparse.py]: done - I had forgotten what this was for  :-S

File: pypit/data/settings/archive/settings.2017-11-17.apf_levy (hunk 4 above)
Comment (rcooke-ast) [reply pypit/data/settings/archive/settings.2017-11-17.apf_levy]: all settings (and their archived versions) are updated.

**Comment by profxj** [comment]
merging.

**Pull request merged**